src/puppetlabs/puppetdb/query/population.clj

```diff
@@ -2,29 +2,52 @@
   "Population-wide queries
 
    Contains queries and metrics that apply across an entire population."
-  (:require [puppetlabs.puppetdb.jdbc :refer [query-to-vec table-count with-transacted-connection]]
+  (:require [puppetlabs.puppetdb.jdbc :refer [query-to-vec
+                                              table-count
+                                              with-transacted-connection]]
             [puppetlabs.kitchensink.core :refer [quotient]]
-            [metrics.gauges :refer [gauge]]))
+            [metrics.gauges :refer [gauge]]
+            [honeysql.core :as hcore]
+            [honeysql.helpers :as hh]))
+
+;;; Query library
+
+(def ^:private from-all-resources-and-their-nodes
+  (hcore/build :from [:catalogs :catalog_resources :certnames]
+               :merge-where [:and
+                             [:= :catalogs.id :catalog_resources.catalog_id]
+                             [:= :certnames.certname :catalogs.certname]]))
+
+(defn- where-nodes-are-active [q]
+  (hh/merge-where q [:and
+                     [:= :certnames.deactivated nil]
+                     [:= :certnames.expired nil]]))
+
+(defn- select-count [q]
+  (-> q
+      (hh/select [:%count.* :c])
+      hcore/format
+      query-to-vec
```

Comment:
The `(comp hcore/format query-to-vec)` might make a good helper in `puppetlabs.puppetdb.honeysql`